### PR TITLE
Add HOBA sample to the www-authenticate index.md

### PR DESCRIPTION
**Pull request by granowski-bamboo**
This change is to add a sample HOBA scheme representation for developers to see.

There is another scheme out there for authentication with HTTP. It is defined here:
https://datatracker.ietf.org/doc/html/rfc7486

### Motivation
In order to facilitate use of HOBA auth with developers, I figured it would be a good opportunity to add a sample of what HOBA looks like for WWW-Authenticate and Authorization. As it was easily misunderstood by coworkers trying to extract it from the RFC.

### CODE REVIEW

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 1]
```diff
@@ -135,6 +135,16 @@ Generally you will need to check the relevant specifications for these (keys for
 - `userhash` {{optional_inline}}
   - : A server may specify `"true"` to indicate that it supports username hashing (default is `"false"`)
 
+### HOBA
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/www-authenticate/index.md:138]:
```suggestion
### HTTP Origin-Bound Authentication (HOBA)
```

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 2]
```diff
@@ -135,6 +135,16 @@ Generally you will need to check the relevant specifications for these (keys for
...
+
+- `<challenge>`
+  - : A sequence of fields concatenated into a string in the format of '\<len\>:\<value\>'.
+    A to-be-signed string that is defined by RFC 7486.
+- `<max-age>`
+  - : The number of seconds from the time the HTTP response is emitted for which responses to this challenge can be accepted.
+- `realm` {{optional_inline}}
+  - : The scope of protection in the manner described in HTTP/1.1, Authentication RFC7235
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/www-authenticate/index.md:146]: So generally we don't want to link to specs for detail, but in MDN use human-readable text. 

You could duplicate the text from above, or link to it. This is what is shown above.

```
- **realm=**\<realm> {{optional_inline}}
  - : A string describing a protected area.
    A realm allows a server to partition up the areas it protects (if supported by a scheme that allows such partitioning), and informs users about which particular username/password are required.
    If no realm is specified, clients often display a formatted hostname instead.
```

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 3]
```diff
@@ -135,6 +135,16 @@ Generally you will need to check the relevant specifications for these (keys for
 - `userhash` {{optional_inline}}
   - : A server may specify `"true"` to indicate that it supports username hashing (default is `"false"`)
 
+### HOBA
+
+- `<challenge>`
+  - : A sequence of fields concatenated into a string in the format of '\<len\>:\<value\>'.
+    A to-be-signed string that is defined by RFC 7486.
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/www-authenticate/index.md:142]: Again, can this be summarised in human readable form?

**Comment by hamishwillee** [comment]
Thanks very much @granowski-bamboo 

Looks good to me, modulo comments, and confirmation from @teoli2003 

What's missing is browser compatibility and specification links. These would both be defined in browser compatibility data https://github.com/mdn/browser-compat-data/blob/main/http/headers/WWW-Authenticate.json

That's what provides the specification and browser compatibility info at the bottom of the page: https://pr27852.content.dev.mdn.mozit.cloud/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#specifications

Do you have information about what browsers and versions it is supported in?

**Comment by granowski-bamboo** [comment]
@hamishwillee I don't have that info right now, but I'll work on these PR feedback items as soon as I can.
I didn't think anyone would even respond to this PR really... so yeah, give me a couple days to make sure I can gather the info since I'm balancing work life and this little extra contribution attempt.

File: files/en-us/web/http/headers/www-authenticate/index.md (hunk 2 above)
Comment (granowski) [reply files/en-us/web/http/headers/www-authenticate/index.md]: I opted for linking to the directives section where realm is defined as the definition is the same as defined elsewhere. I used an anchor link so if someone clicks the link it'll take them right to the part of the page. It'll also keep people thinking about realms as a general HTTP auth thing.

File: files/en-us/web/http/headers/www-authenticate/index.md (hunk 3 above)
Comment (granowski) [reply files/en-us/web/http/headers/www-authenticate/index.md]: I believe my new commit should suffice. If the developer needs to know a bit more about how it's concatenated and any details of the individual parts, they really should go to the RFC or look at the sample I added in the examples section.
Will the latest work?

**Comment by granowski** [comment]
@hamishwillee I see I need to fork the browser-compat-data repository as well. I've started to do that in order to get the Specifications and Compatibility section updated.

File: files/en-us/web/http/headers/www-authenticate/index.md (hunk 2 above)
Comment (hamishwillee) [reply files/en-us/web/http/headers/www-authenticate/index.md]: Works for me!

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 4]
```diff
@@ -73,7 +73,7 @@ WWW-Authenticate: Basic realm=<realm>
 WWW-Authenticate: Basic realm=<realm>, charset="UTF-8"
 ```
 
-## Directives
+## [Directives](#directives)
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/www-authenticate/index.md:76]: We don't need to do this (and we shouldn't). MDN automatically creates an anchor link for every heading. This is a lower cased version with underscores replacing spaces. If you hover over a heading you will see the `#` (left of heading) which you can copy for the link/anchor in more complicated cases than this one.

image


```suggestion
## Directives
```

**Comment by hamishwillee** [comment]
> @hamishwillee I see I need to fork the browser-compat-data repository as well. I've started to do that in order to get the Specifications and Compatibility section updated.

Thanks @granowski-bamboo . 

That's awesome. I'm out of office 2 days every week so if you don't get immediate replies that's why.

We should get review from @teoli2003 too.

**Review by teoli2003 — APPROVED** [review]
👍

**Comment by granowski-bamboo** [comment]
Hey, thanks for merging! I'll continue to dig into the browser-compat-data repository for updates there. May be a day or two before I get anything up for that though. We're just getting into my works next sprint work.

**Comment by hamishwillee** [comment]
@granowski-bamboo Thanks - looking forward to the BCD updates when you have a moment.

**Comment by granowski** [comment]
@hamishwillee I've started the draft here: https://github.com/mdn/browser-compat-data/pull/20370

So far I have the json structure in that file, but given that the WWW-Authenticate header has always been acceptable by browsers since general processing of HTTP headers has existed, I'm not sure what versions I should be putting in the json structure. I checked it in with nulls under the assumption that null means the doc page will render "unknown compatibility" at the moment.

**Comment by hamishwillee** [comment]
Thanks very much. I've commented there.

**Pull request merged**